catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/forms/filter/TransformVisitor.java

```diff
@@ -77,6 +78,35 @@ public void visitLiteralType(VisitableElement<List<Serializable>> visitable) {
     builder.setValue(values.get(0).toString());
   }
 
+  @Override
+  public void visitDistanceType(VisitableElement<Double> visitable) {
+    Double value = visitable.getValue();
+    if (value == null) {
+      LOGGER.debug("No values found on distance type");
+      return;
+    }
+
+    builder.setDistance(value);
+  }
+
+  @Override
+  public void visitDistanceBufferType(VisitableElement<List<Object>> visitable) {
+    traceName(visitable);
+    List<Object> values = visitable.getValue();
+    if (CollectionUtils.isEmpty(values)) {
+      LOGGER.debug("No values found on distance buffer type");
+      return;
+    }
+
+    builder.beginBinarySpatialType("DWITHIN");
+    visitable
+        .getValue()
```

Comment:
✏️ `filter(Objects::nonNull)`

Reply:
If an element in the list is null, the `forEachOrdered(v -> v.accept(this))` call will fail with an NPE, which would indicate a programmer error that should be fixed upstream so I'm going to leave this as is. The user would probably see the red error box in that case.